src/main/java/org/jenkinsci/plugins/github/config/GitHubPluginConfig.java

```diff
@@ -90,8 +90,15 @@ public void setConfigs(List<GitHubServerConfig> configs) {
         return configs;
     }
 
+    /**
+     * @deprecated get GitHubServerConfig matching to exact remote GH server, filter and only then check allowToManage
```

Comment:
nope, we can skip some actions if no any server is allowed to manage hooks (such as cleanup)